mpc/src/common/mod.rs

```diff
@@ -79,6 +81,52 @@ pub fn lagrange_interpolate<F: FftField>(
         return Err(ShareError::InvalidInput);
     }
     let n = x_vals.len();
+
+    // For small inputs, use sequential computation to avoid thread overhead
+    if n <= 4 {
```

Comment:
Do we have an empirical way to see what n should actually be here? Like an elbow curve?